### PR TITLE
feature - createAddStream

### PR DESCRIPTION
**Pull request by diasdavid**
- [x] fix adding empty dirs
- [x] createAddStream tests on interface-ipfs-core
- [x] createAddStream implemented

### CODE REVIEW

**Comment by diasdavid** [comment]
- [ ] need to add the promisify

File: src/api/add-stream.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        res.forEach((tuple) => {
+          ds.push(tuple)
+        })
+
+        ds.push(null)
+      })
+    }
+    callback(null, ds)
+  })
```
Comment (haadcode) [inline src/api/add-stream.js]: Are errors handled somewhere?

**Comment by haadcode** [comment]
One question re. error handling, otherwise LGTM

File: src/api/add-stream.js (hunk 1 above)
Comment (diasdavid) [inline src/api/add-stream.js]: good catch, tahnk you. Emitted the error upstream :)

**Pull request merged**